.circleci/config.yml

```diff
@@ -1,5 +1,43 @@
 version: 2.1
 commands:
+  polymetis_tests:
+    steps:
+      - run:
+          name: Run polymetis pytest
+          command: |
+            . /opt/conda/etc/profile.d/conda.sh
+            conda activate polymetis-local
+
+            Xvfb :100 -screen 0 1024x768x24 &
+            DISPLAY=:100 pytest ./polymetis/polymetis/tests/python
+            pkill Xvfb
+      - run:
+          name: Run polymetis C++ unit tests
+          command: |
+            . /opt/conda/etc/profile.d/conda.sh
+            conda activate polymetis-local
+
+            export PATH=$PWD/polymetis/polymetis/build/:$PATH
+
+            python ./polymetis/polymetis/tests/cpp/test_server.py
+      - run:
+          name: Run integration tests (empty client)
+          command: |
+            . /opt/conda/etc/profile.d/conda.sh
+            conda activate polymetis-local
+            ./polymetis/tests/test_empty_statistics_client.sh
+      - run:
+          name: Run integration tests (mocked hardware client)
+          command: |
+            . /opt/conda/etc/profile.d/conda.sh
+            conda activate polymetis-local
+            ./polymetis/tests/test_mocked_hardware_client.sh
+      - run:
+          name: Run integration tests (simulation)
+          command: |
+            . /opt/conda/etc/profile.d/conda.sh
+            conda activate polymetis-local
+            ./polymetis/tests/test_simulation.sh
```

Comment:
What's the motivation behind splitting `test_integrations.sh` into multiple scripts? I agree it looks much cleaner, but it also results in some duplicated code in the scripts and also a much longer `polymetis_tests` shared command.

Reply:
I found it exceedingly difficult to figure out what was going on when the integration tests broke -- there's a ton of printout, several error messages from killing the server repeatedly, and in general hard to pinpoint what was going on. For example, here I was testing to make sure the tests actually broke when the multithread wasn't fixed; I was expect it should only break the simulation tests.

I do agree it's repetitive -- any thoughts on how to do it while preserving clarity?

Reply:
Some ideas:
1. Minimize printouts so as to not dilute the error messages, maybe have a consistent way of configuring verbosity so running integration tests won't have the same verbosity with a user running each process in a separate terminal.
2. Have the server exit gracefully upon an interrupt and have the test script terminate the server with signals instead of pkill-ing.
3. Pass any exception on the server side as a string field back to the user and have better error handling on the client side so it's easy to pinpoint when the exception happened from the error msgs.

Until we get there, I agree it would make things easier by keeping the scripts separate, and minor tech debt in the repetitiveness can be easily removed afterwards.

Reply:
Added this as an issue in #930 for TODO post fbrp integration